### PR TITLE
21 stubs on the wall, take one down, or do them all...

### PR DESCRIPTION
**Pull request by seanrogers2657**
Compiles fine. Got some comments.

### CODE REVIEW

File: parser.cs [hunk 1]
```diff
@@ -449,86 +449,479 @@ public class Parser {
...
+                // Rule 106
+                functionIdentifier();
+                optionalActualParameterList();
+                break;
+            case TOKENS.INTEGER_LIT:
+                // unsigned integer? Rule 99?
+                match(TOKENS.INTEGER_LIT);
+                break;
+            case TOKENS.FIXED_LIT:
+                // Question mark on table? rule 100? not sure
```
Comment (seanrogers2657) [inline parser.cs]: There was a question mark in the table for factor(). Not sure why but it's worth noting.

Comment (brianpmaher) [inline parser.cs]: Defer to @JesseStephenBrown

File: parser.cs [hunk 2]
```diff
@@ -449,86 +449,479 @@ public class Parser {
...
+                // Rule 70
+                match(TOKENS.COMMA);
+                actualParameter();
+                actualParameterTail();
+                break;
+            case TOKENS.RPAREN:
+                // Rule 71
+                break;
+            default:
+                error("An error with actualParameterTail");
```
Comment (brianpmaher) [inline parser.cs:463]: These errors should be more descriptive, but don't worry about them on MS4, I added a ticket for this in MS5.

**Comment by brianpmaher** [comment]
Looks good, +1. Get feedback from @JesseStephenBrown on line 784 before you merge.

File: parser.cs (hunk 1 above)
Comment (JesseStephenBrown) [inline parser.cs]: Oh, so for some reason the grammar doesn't specify that factor should go to fixed lit, but it totes m'goats should. So just match the token that is in the case for float and fixed and it should be fine.

File: parser.cs (hunk 2 above)
Comment (JesseStephenBrown) [inline parser.cs:463]: Yeah, errors can be dealt with later. We're ahead of the game boys. What goes around is all around.

**Comment by JesseStephenBrown** [comment]
Nice use of rule comments, makes things easy to check. +1 after the line 789 is changed

**Pull request merged**